src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Unix.cs

```diff
@@ -530,5 +535,112 @@ public static string[] GetLogicalDrives()
         {
             return DriveInfoInternal.GetLogicalDrives();
         }
+
+        /// <summary>Gets the path of the target of the specified link.</summary>
+        /// <param name="linkPath">A path to a link file.</param>
+        /// <param name="isDirectory">Whether the link represents a directory or not. Irrelevant in Unix since readlink does not care about the underlying type.</param>
+        /// <returns>If linkPath represents a link file and it exists, returns the link's target path.
+        /// If linkPath is not a link or the target does not exist, returns null.</returns>
+        internal static string? GetLinkTarget(ReadOnlySpan<char> linkPath, bool isDirectory) => Interop.Sys.ReadLink(linkPath);
+
+        /// <summary>
+        /// Creates a file symbolic link identified by path that points to pathToTarget.
+        /// </summary>
+        /// <param name="path">The path where the symbolic link should be created.</param>
+        /// <param name="pathToTarget">The path of the target to which the symbolic link points.</param>
+        /// <param name="isDirectory">True if the pathToTarget represents a directory or a symlink to a directory.</param>
+        internal static void CreateSymbolicLink(string path, string pathToTarget, bool isDirectory)
+        {
+            // Fail if the target exists but is not consistent with the expected filesystem entry type
+            if (Interop.Sys.LStat(pathToTarget, out Interop.Sys.FileStatus targetInfo) == 0)
```

Comment:
Why not use `stat` here to look at the final target?

Reply:
You're correct, it should be `stat`, otherwise we would follow `pathToTarget` if it also happens to be a symlink.

Reply:
My bad, I understood the documentation backwards the last time I read it. It should be `lstat` because I don't want to follow symlinks, I want information about the link itself.
https://man7.org/linux/man-pages/man2/lstat.2.html

> _ lstat() is identical to stat(), except that if pathname is a symbolic link, then it returns information about the link itself, not the file that the link refers to._

Reply:
This is the difference:
```
link -> file
link2link -> link
```
With `stat` creation of `link2link` with `isDirectory` will fail because the final target is not a directory.
With `lstat` it will succeed because only `link` is checked.

The `isDirectory` field also uses `stat` (eventually):

https://github.com/dotnet/runtime/blob/d87b78342c5d88741eb1f194b4c9968baf192ddc/src/libraries/System.Private.CoreLib/src/System/IO/FileStatus.Unix.cs#L360-L369

Reply:
I agree with @tmds, it should be `stat` instead.